### PR TITLE
resetcause:restore to factory, factory to factory_inquiry

### PR DESCRIPTION
**Pull request by gneworld**
## Summary
 add restore to indicate "restore the factory settings" and  the previous "factory" represents the confirmation of restoring to factory settings.
## Impact
 resetcause
## Testing
 CI test

### CODE REVIEW

**Review by cederom — CHANGES_REQUESTED** [review]
* I like the idea but this changes existing logic and may impact existing applications. 
* After introducing `restore` existing applications may not catch the `factory` event and skip the factory defaults reset.
* `factory` is already used to trigger the "perform reset to factory defaults".
* After successful factory defaults reset one more reset can be triggered with `NULL`  and device should land in a state that should be handled by a firmware accordingly based on existing configuration.
* `restore` name is a bit ambiguous and may be confusing, its not self explanatory, maybe another keyword can be used?
* Maybe something like `defaults` would fit here better? `factory` as before would cause factory defaults reset, and then call reset with `defaults` mark to note everything is default or needs to be set default (this state should be also unambiguous)?
* Documentation needs an update in this area. Will try to work more with documentation in a free moment.
* Long story short if new resetflag is to create new and unique state that would be fine, but if it changes existing behavior this will break compatibility and we do not want that.
* I would like to hear what other devs think about this idea :-)

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> * I like the idea but this changes existing logic and may impact existing applications.
> * After introducing `restore` existing applications may not catch the `factory` event and skip the factory defaults reset.
> * `factory` is already used to trigger the "perform reset to factory defaults".
> * After successful factory defaults reset one more reset can be triggered with `NULL`  and device should land in a state that should be handled by a firmware accordingly based on existing configuration.
> * `restore` name is a bit ambiguous and may be confusing, its not self explanatory, maybe another keyword can be used?
> * Maybe something like `defaults` would fit here better? `factory` as before would cause factory defaults reset, and then call reset with `defaults` mark to note everything is default or needs to be set default (this state should be also unambiguous)?
> * Documentation needs an update in this area. Will try to work more with documentation in a free moment.
> * Long story short if new resetflag is to create new and unique state that would be fine, but if it changes existing behavior this will break compatibility and we do not want that.
> * I would like to hear what other devs think about this idea :-)

@cederom this change is to align with nuttx side definition:
https://github.com/apache/nuttx/blob/master/include/sys/boardctl.h#L461-L471

**Review by acassis — APPROVED** [review]
